### PR TITLE
Cherry-picking ML squad work for v0.23.3

### PR DESCRIPTION
**Pull request by brimoor**
Cherry-picks the following PRs from `develop` onto `release/v0.23.3`:
- https://github.com/voxel51/fiftyone/pull/3977
- https://github.com/voxel51/fiftyone/pull/3983
- https://github.com/voxel51/fiftyone/pull/3986
- https://github.com/voxel51/fiftyone/pull/3990
- https://github.com/voxel51/fiftyone/pull/3991
- https://github.com/voxel51/fiftyone/pull/4003

### CODE REVIEW

File: docs/source/integrations/index.rst
Comment (jacobmarks) [inline docs/source/integrations/index.rst:1]: I like this change. Thoughts on changing `model-zoo` to `models`?

**Review by jacobmarks — DISMISSED** [review]
LGTM!

One note: I think we're missing batch inference for YOLO-NAS — at least I wasn't able to get it to work for me. Aside from that, everything works as it should. Just made some updates to the transformers docs :)

File: docs/source/integrations/index.rst
Comment (brimoor) [reply docs/source/integrations/index.rst:1]: 💯 made the change 👍

**Review by jacobmarks — APPROVED** [review]
LGTM 👍

**Pull request merged**